### PR TITLE
`[Issue #2565]` network config extended

### PR DESCRIPTION
**Pull request by Da-Colon**
Merges with #2595 
Discussed in this comment https://github.com/decentdao/decent-interface/pull/2595#issuecomment-2524193062

This PR will cover any work needed to ensure that the network is properly being used in the App.

### CODE REVIEW

File: src/hooks/utils/useAutomaticSwitchChain.ts [hunk 1]
```diff
@@ -8,11 +9,25 @@ export const useAutomaticSwitchChain = ({
...
+  const { switchChain } = useSwitchChain({
+    mutation: {
+      onError: () => {
+        if (addressPrefix !== urlAddressPrefix && urlAddressPrefix !== undefined) {
+          const chainId = getChainIdFromPrefix(urlAddressPrefix);
+          switchChain({ chainId });
+        }
+      },
+    },
+  });
```
Comment (Da-Colon) [inline src/hooks/utils/useAutomaticSwitchChain.ts:21]: Funny enough, this kinda reverts back to how the file was before only this now we are attempting the switch chain again onError. 

While, not ideal. this does work and fixes the current network switcher issues we've been having. Ideally I think we would have completely seperate `publicClient` by passing the `id` directly in. and then move all network switching to when a user clicks to send a transaction.

But I'm not sure how that scales at the moment and want to discuss further before we make that change.

So this is the current path of least resistance. currently there are two spots that this is going to be required.

- `useAutomaticSwitchChain`
- `EstablishEssentials` where a new network switcher will exist.

**Review by Da-Colon — COMMENTED** [review]
Another. I'll create any other PRs for anything pre-existing issues off develop I find while testing so keep an eye out for those. 

And one PR branched from here to updates the DAO search to be cross-chain per @adamgall suggestion. 

- I have another commit to add the SwitchChain to the network switcher in DAO creation coming and this will be good for other testers starting tomorrow
-

**Comment by Da-Colon** [comment]
Okay, I feel like this is good as its going to get in its current form. There is a flash of the network error screen, but that was always there before, but now it actually loads. I think the state refactor will help this get rid of the flash.

**Review by adamgall — APPROVED** [review]

File: src/providers/NetworkConfig/web3-modal.config.ts [hunk 2]
```diff
@@ -12,29 +12,32 @@ const supportedWagmiChains = supportedNetworks.map(network => network.chain);
...
+  metadata,
   transports: supportedNetworks.reduce(transportsReducer, {}),
   batch: {
     multicall: true,
   },
 });
 
 if (walletConnectProjectId) {
-  createWeb3Modal({ wagmiConfig, projectId: walletConnectProjectId });
+  createWeb3Modal({ wagmiConfig, projectId: walletConnectProjectId, metadata: metadata });
```
Comment (mudrila) [inline src/providers/NetworkConfig/web3-modal.config.ts:42]: Can use prop shorthand here as well

**Review by mudrila — APPROVED** [review]
1 very minor comment, but otherwise - looks good!

**Pull request merged**